internal/authz/github/github.go

```diff
@@ -59,34 +73,148 @@ func (p *Provider) Validate() (problems []string) {
 	return nil
 }
 
-// FetchUserPermsByToken fetches all the private repo ids that the token can
-// access.
-func (p *Provider) FetchUserPermsByToken(ctx context.Context, token string) (*authz.ExternalUserPermissions, error) {
+// FetchUserPermsByToken fetches all the private repo ids that the token can access.
+//
+// This may return a partial result if an error is encountered, e.g. via rate limits.
+func (p *Provider) FetchUserPermsByToken(ctx context.Context, token string, opts *authz.FetchPermOptions) (*authz.ExternalUserPermissions, error) {
 	// 🚨 SECURITY: Use user token is required to only list repositories the user has access to.
 	client := p.client.WithToken(token)
 
 	// 100 matches the maximum page size, thus a good default to avoid multiple allocations
 	// when appending the first 100 results to the slice.
-	repoIDs := make([]extsvc.RepoID, 0, 100)
+	const repoSetSize = 100
+	perms := &authz.ExternalUserPermissions{
+		Exacts: make([]extsvc.RepoID, 0, repoSetSize),
+	}
+	seenRepos := make(map[extsvc.RepoID]bool, repoSetSize)
```

Comment:
```suggestion
	seenRepos := make(map[extsvc.RepoID]struct{}, repoSetSize)
```

https://dave.cheney.net/2014/03/25/the-empty-struct